src/site/_filters/filter-by-lang.js

```diff
@@ -1,11 +1,25 @@
+const {defaultLocale} = require('../_data/site');
+const {getDefaultUrl} = require('../_filters/urls');
 /**
  * Filters-in posts that have the same lang attribute as requested.
- * @param {Array} posts An array of posts.
- * @param {string} lang Target language.
+ * @param {EleventyCollectionItem[]} posts An array of posts.
+ * @param {string} [lang] Target language.
+ * @return {EleventyCollectionItem[]}
  */
-module.exports = function filterByLang(posts, lang) {
-  if (!lang) {
-    return posts;
+module.exports = function filterByLang(posts, lang = defaultLocale) {
+  /** @type Map<string, EleventyCollectionItem> */
+  const filteredPosts = new Map();
+  for (const post of posts) {
+    if (post.data && [lang, defaultLocale].includes(post.data.lang)) {
+      const defaultUrl = getDefaultUrl(post.url);
+      if (filteredPosts.has(defaultUrl) && post.data.lang === lang) {
+        filteredPosts.set(defaultUrl, post);
+      } else if (!filteredPosts.has(defaultUrl)) {
+        filteredPosts.set(defaultUrl, post);
```

Comment:
Before the filter would filter only posts belonging to the language passed in. As a result, if we applied the spanish filter, only spanish articles would show up. I think we've since decided we want to show english (default language) articles and if there is a version of the article in the selected language then we show that one instead of the english version.

So basically this function now shows all articles of the default language, except when we have a translated article, then we display that one instead.